.github/workflows/build-docker-image-and-binaries.yaml

```diff
@@ -36,58 +38,82 @@ jobs:
           submodules: recursive
 
       - name: Install Cosign
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         uses: sigstore/cosign-installer@main
 
-      - name: Import GPG key
-        id: import_gpg
-        uses: crazy-max/ghaction-import-gpg@v4
-        with:
-          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
-          passphrase: ${{ secrets.GPG_PASSPHRASE }}
-
       - name: Set up QEMU
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         uses: docker/setup-qemu-action@v1
 
       - name: Set up Docker Buildx
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         id: buildx
         uses: docker/setup-buildx-action@v1
         with:
           config: .github/buildkit-config.toml
 
       - name: Login to DockerHub
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         uses: docker/login-action@v1
         with:
           username: ${{ secrets.DOCKER_HUB_USERNAME }}
           password: ${{ secrets.DOCKER_HUB_TOKEN }}
 
       - name: Build and push docker image for all platforms
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         run: |
           make build-push-docker-images
         env:
           SEMVER: ${{ github.event.inputs.tag }}
 
       - name: Sign the images with GitHub OIDC
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
         run: cosign sign --oidc-issuer https://token.actions.githubusercontent.com ${TAGS}
         env:
           TAGS: axelarnet/axelar-core:${{ github.event.inputs.tag }}
           COSIGN_EXPERIMENTAL: 1
 
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
       - name: Build Binaries for Linux/MacOS
         env:
           SEMVER: ${{ github.event.inputs.tag }}
         run: |
-          make build-binaries-in-docker
+          if [ "$RUNNER_OS" == "Linux" ]
+          then
+              OS="linux"
+          elif [ "$RUNNER_OS" == "MacOS" ]
+          then
+              OS="darwin"
+          fi
+          ARCH="${{ matrix.arch }}"
+          make build
```

Comment:
`make build` is a bit different than `make build-binaries`. In the latter the build command that runs is
`go build -o ./bin/axelard-"$os"-"$arch"-"$version" -mod=readonly -tags "$build_tags" -ldflags "$ldflags" ./cmd/axelard`
Can you update the build command in `Makefile` for `make build` to include all those args?

Reply:
We should be ok without modifying the build command in the Makefile.
for "build" in the Makefile we have : go build -o ./bin/axelard -mod=readonly $(BUILD_FLAGS) ./cmd/axelard 
$(BUILD_FLAGS)  regroups tags and ldflags, see https://github.com/axelarnetwork/axelar-core/blob/main/Makefile#L17

I also changed the logic for the binary rename, it is done in gh actions directly no more in the script see https://github.com/axelarnetwork/axelar-core/blob/feat/build-matrix/.github/workflows/build-docker-image-and-binaries.yaml#L87-L100

Reply:
Ah, I probably read that as `BUILD_TAGS` instead.